graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
+		return fmt.Errorf("pulling with digest reference failed from v2 registry, not falling back to v1")
```

Comment:
I agree in general, but this is surrounded by log messages such as:
```
logrus.Errorf("Error from V2 registry: %s", err)
logrus.Debug("image does not exist on v2 registry, falling back to v1")
logrus.Debugf("pulling v1 repository with local name %q", repoInfo.LocalName)
```
However, would this be a reasonable error message:
> secure pull with digest reference failed
Or should I just remove everything after the comma in the original message?